dcicutils/s3_utils.py

```diff
@@ -24,16 +25,70 @@ class s3Utils(object):  # NOQA - This class name violates style rules, but a lot
     RAW_BUCKET_TEMPLATE = "elasticbeanstalk-%s-files"
     BLOB_BUCKET_TEMPLATE = "elasticbeanstalk-%s-blobs"
 
-    def __init__(self, outfile_bucket=None, sys_bucket=None, raw_file_bucket=None,
-                 blob_bucket=None, env=None):
+    @staticmethod
+    def verify_and_get_env_config(s3_client, global_bucket: str):
+        """ Verifies the S3 environment from which the env config is coming from, and returns the S3-based env config
+            Throws exceptions if the S3 bucket is unreachable, or an env based on the name of the global S3 bucket
+            is not present.
         """
-        if we pass in env set the outfile and sys bucket from the environment
+        head_response = s3_client.head_bucket(Bucket=global_bucket)
+        status = head_response['ResponseMetadata']['HTTPStatusCode']  # should be 200; raise error for 404 or 403
+        if status != 200:
+            raise Exception('Could not access GLOBAL_BUCKET_ENV {global_bucket}: status: {status}'.format(
+                global_bucket=global_bucket, status=status))
+        # list contents of global bucket, look for a match with the global bucket name
+        list_response = s3_client.list_objects_v2(Bucket=global_bucket)
+        # no match, raise exception
+        if list_response['KeyCount'] < 1:
+            raise Exception('No config objects found in global bucket {global_bucket}'.format(
+                global_bucket=global_bucket))
+        keys = [list_response['Contents'][i]['Key'] for i in range(0, len(list_response['Contents']))]
+        config_filename = None
+        for filename in keys:
+            # multiple matches, raise exception
+            if filename in global_bucket and config_filename is not None:
+                raise Exception('multiple matches for global env bucket: {global_bucket}; keys: {keys}'.format(
```

Comment:
Here again (untested, and the last one I'll spell out):
```
class GlobalBucketMultipleMatches(GlobalBucketException):
    MESSAGE = "Multiple matches for global env bucket: {global_bucket}; keys: {keys}"
    def __init__(self, bucket, keys):
        super().__init__(bucket=bucket, keys=keys)
```
so that here in context we can do (yup, still untested):
```
    raise GlobalBucketMultipleMatches(bucket=global_bucket, keys=keys)
```
and so on for the other exceptions below.